packages/ui/src/components/Visualization/Canvas/Canvas.tsx

```diff
@@ -57,14 +58,78 @@ export const Canvas: FunctionComponent<PropsWithChildren<CanvasProps>> = ({ enti
   /** Context to interact with the Canvas catalog */
   const catalogModalContext = useContext(CatalogModalContext);
 
-  const controller = useMemo(() => ControllerService.createController(), []);
+  const controller = useVisualizationController();
```

Comment:
It was looking like was redrawing a few times.  Could it be caused by the removal of this useMemo?

Reply:
Thanks for checking @dlabaj, that change was to mimic as much as possible the demo application, so I could compare 1:1 with the hope of understanding what's causing the flickering.

The `useMemo` is still in the code but in the [`CanvasController`](https://github.com/KaotoIO/kaoto/pull/1469/files#diff-e7c41c723cf815e75a0a33cb6281dbb0876a1ace6492d9391c02f57be3790a55R14) component:

```typescript
export const CanvasController: FunctionComponent<PropsWithChildren<CanvasProps>> = ({ entities }) => {
  const controller = useMemo(() => ControllerService.createController(), []);

  return (
    <VisualizationProvider controller={controller}>
      <Canvas entities={entities} /> // This is the same component from where the useMemo was removed
    </VisualizationProvider>
  );
};
```

Reply:
ah okay.  Thanks @lordrip .